CODEOWNERS

```diff
@@ -0,0 +1,22 @@
+#
+# CODEOWNERS is a GitHub standard to specify who is automatically assigned pull requests to review. 
+# This helps to prevent pull requests from languishing without review.
+# GitHub can also be configured to require review from code owners before a pull request can be merged.
+#
+# Further reading is available from the following two URLs:
+#   https://blog.github.com/2017-07-06-introducing-code-owners/
+#   https://help.github.com/articles/about-codeowners/  
+#
+# It uses the same pattern rule for gitignore file
+# https://git-scm.com/docs/gitignore#_pattern_format
+#
+
+# Any pull request containing changes of files inside a 'data-plane' or 'client' directory will be reviewed 
+# by the azure-java-client-sdk-team.
+# Because there is a restructure coming up, and because 'client' is an overloaded term, there is a single
+# wildcard star being used, so we only match one directory, not many.
+*/data-plane/**       @azure/azure-java-client-sdk-team
```

Comment:
Are we able to have multiple owners per rule? I was almost thinking we'd have an explicit list of each and every package with explicit owners on them.

Reply:
Yes, we can have multiple owners listed against a rule. We can use explicit lists rather than teams. My thinking was mainly around management - as the teams change, it might be easier to edit the team list in GitHub, rather than update CODEOWNERS. Happy to do it either way though.

Reply:
Lets talk some more about it. I agree with the maintenance benefits. That said, I was hoping that it would help reduce noise for the team and get the right reviewers on PR's.

Reply:
Yes - having a more specific assignee per SDK component will definitely be less noisy. I'll revise this PR after some discussion with you and others over long-term plans for ownership of each SDK component (I probably shouldn't assume right now that the current porters into the mono-repo are necessarily the designated long term support people...)